### PR TITLE
feat(dbt): add reference for collecting columns metadata using the `dagster` dbt package

### PR DESCRIPTION
**Pull request by rexledesma**
## Summary & Motivation
Rework https://github.com/dagster-io/dagster/pull/19551 by using https://github.com/dagster-io/dagster/pull/19623 as the mechanism for collecting dbt metadata. 

## How I Tested These Changes
read, try it local

### CODE REVIEW

**Comment by rexledesma** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/19628?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `master`
  * **PR #19545** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/19545?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #19546** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/19546?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #19547** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/19547?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #19623** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/19623?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #19628** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/19628?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
        * **PR #19548** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/19548?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #19551** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/19551?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/19628?utm_source=stack-comment).

File: docs/content/integrations/dbt/reference.mdx [hunk 1]
```diff
@@ -532,6 +532,39 @@ def my_dbt_assets(context: AssetExecutionContext, dbt: DbtCliResource):
 
 ---
 
+## Emit dbt table schema as materialization metadata <Experimental />
```
Comment (sryza) [inline docs/content/integrations/dbt/reference.mdx:535]: Nitpick: I think calling this "column schema" or "column metadata" would be more clear. "Table schema" could be interpreted as "the database schema that the table lives inside".

File: docs/content/integrations/dbt/reference.mdx [hunk 2]
```diff
@@ -532,6 +532,39 @@ def my_dbt_assets(context: AssetExecutionContext, dbt: DbtCliResource):
...
+
+Dagster allows you to emit table schema [materialization metadata](/concepts/assets/software-defined-assets#recording-materialization-metadata), which includes the column names and data types of your materialized dbt models, seeds, and snapshots.
+
+With this metadata, you can view documentation for all columns and not just columns described in your dbt project.
+
+To collect this metadata for your dbt project, specify the `dagster` dbt package in your `dependencies.yml`, and then run `dbt deps`.
```
Comment (sryza) [inline docs/content/integrations/dbt/reference.mdx:541]: Does this mean that they'll also need to run dbt dips in CI?

File: docs/content/integrations/dbt/reference.mdx [hunk 3]
```diff
@@ -532,6 +532,39 @@ def my_dbt_assets(context: AssetExecutionContext, dbt: DbtCliResource):
 
 ---
 
+## Emit dbt table schema as materialization metadata <Experimental />
+
+Dagster allows you to emit table schema [materialization metadata](/concepts/assets/software-defined-assets#recording-materialization-metadata), which includes the column names and data types of your materialized dbt models, seeds, and snapshots.
+
+With this metadata, you can view documentation for all columns and not just columns described in your dbt project.
```
Comment (sryza) [inline docs/content/integrations/dbt/reference.mdx:539]:
```suggestion
With this metadata, you can view documentation in Dagster for all columns, not just columns described in your dbt project.
```

File: docs/content/integrations/dbt/reference.mdx [hunk 4]
```diff
@@ -532,6 +532,39 @@ def my_dbt_assets(context: AssetExecutionContext, dbt: DbtCliResource):
...
+
+To collect this metadata for your dbt project, specify the `dagster` dbt package in your `dependencies.yml`, and then run `dbt deps`.
+
+```yaml
+packages:
+  - git: "https://github.com/dagster-io/dagster.git"
+    subdirectory: "/python_modules/libraries/dagster-dbt/packages/dagster"
+    revision: {{ DAGSTER_VERSION }} # replace with the version of `dagster` you are using.
```
Comment (sryza) [inline docs/content/integrations/dbt/reference.mdx:547]: Does this mean that whenever they update their Dagster version they'll need to remember to come here and update this?

File: docs/content/integrations/dbt/reference.mdx (hunk 2 above)
Comment (rexledesma) [reply docs/content/integrations/dbt/reference.mdx:541]: Yeah. Running any command (e.g. `dbt build`, `dbt compile`) will fail unless `dbt deps` is run. We already recommend running `dbt deps`: https://docs.dagster.io/integrations/dbt/reference#deploying-a-dagster-project-with-a-dbt-project.

File: docs/content/integrations/dbt/reference.mdx (hunk 4 above)
Comment (rexledesma) [reply docs/content/integrations/dbt/reference.mdx:547]: Not necessarily. If we maintain backcompat with the serialized format that we output in the macro, then users should only need to update if we add additional metadata from this macro that we want to surface in the UI.

File: docs/content/integrations/dbt/reference.mdx [hunk 5]
```diff
@@ -532,6 +532,39 @@ def my_dbt_assets(context: AssetExecutionContext, dbt: DbtCliResource):
 
 ---
 
+## Emit dbt column schema as materialization metadata <Experimental />
```
Comment (sryza) [inline docs/content/integrations/dbt/reference.mdx:535]: Does this heading need the word "dbt" in it?

**Review by sryza — COMMENTED** [review]
Not necessarily in this PR, but should we add this to our template?

**Comment by rexledesma** [comment]
> Not necessarily in this PR, but should we add this to our template?

This new approach involves editing the dbt project to enable this feature. We don't have control over the dbt project in the template CLI (dagster-dbt project scaffold)

We could add a link to the reference after we're completed scaffolding though!

**Comment by sryza** [comment]
> We don't have control over the dbt project in the template CLI (dagster-dbt project scaffold)

Ah good point

File: docs/content/integrations/dbt/reference.mdx (hunk 5 above)
Comment (rexledesma) [reply docs/content/integrations/dbt/reference.mdx]: Nope, I removed it. Column schema is not specific to dbt.

**Pull request merged**